src/dsp_tools/commands/xmlupload/stash/upload_stashed_resptr_props.py

```diff
@@ -31,25 +33,74 @@ def upload_stashed_resptr_props(
 
     print(f"{datetime.now()}: Upload the stashed resptrs...")
     logger.info("Upload the stashed resptrs...")
+
     upload_state.pending_stash = cast(Stash, upload_state.pending_stash)
     link_value_stash = cast(LinkValueStash, upload_state.pending_stash.link_value_stash)
-    for res_id, stash_items in link_value_stash.res_2_stash_items.copy().items():
-        res_iri = upload_state.iri_resolver.get(res_id)
+
+    _upload_process(link_value_stash=link_value_stash, iri_resolver=upload_state.iri_resolver, con=con, context=context)
+    _remove_empty_ids(link_value_stash=link_value_stash)
+
+
+def _upload_process(
+    link_value_stash: LinkValueStash,
+    iri_resolver: IriResolver,
+    con: Connection,
+    context: dict[str, str],
+) -> None:
+    """
+    Try to upload the stashed items. If the target_iri of the item can be found
+    and the item can then successfully be uploaded it is removed from link_value_stash.
+    The link_value_stash is updated accordingly, as a side effect.
+
+    Args:
+        link_value_stash: the current state of the link value stash
+        iri_resolver: service for resolving internal IDs to IRIs
+        con: connection to DSP
+        context: the JSON-LD context of the resource
+    """
+    for res_id, stash_items in link_value_stash.res_2_stash_items.items():
+        res_iri = iri_resolver.get(res_id)
         if not res_iri:
             # resource could not be uploaded to DSP, so the stash cannot be uploaded either
             # no action necessary: this resource will remain in nonapplied_resptr_props,
             # which will be handled by the caller
             continue
         print(f"{datetime.now()}:   Upload resptrs of resource '{res_id}'...")
         logger.info(f"  Upload resptrs of resource '{res_id}'...")
-        for stash_item in stash_items:
```

Comment:
The simplest thing I can do to make your test pass is replacing this line with:

```
for stash_item in reversed(stash_items):
```

When iterating over a list and removing elements, the indices of the remaining elements are shifted. This can cause elements to be skipped or result in accessing invalid indices.

By reversing the list and iterating backwards, the indices of the elements that have not yet been processed are not affected by the removal of elements that have already been processed. This ensures that each element is properly accessed and handled.

All the other code changes in this PR seem to be unnecessary.

Reply:
Yes, you are right I did not understand what was going on, that is why I tried to split the method so that it makes more sense to me what was happening. But since it is about a fix, I guess something like that would be a refactoring task and better be discussed first if it should be done and how.